gobblin-yarn/src/main/java/gobblin/yarn/YarnHelixUtils.java

```diff
@@ -199,4 +206,48 @@ public static void addFileAsLocalResource(FileSystem fs, Path destFilePath, Loca
 
     return environmentVariableMap;
   }
-}
+
+  public static Optional<Integer> takePort(int desiredPort) {
```

Comment:
However, if someone accidentally puts `${PORT_1}` then this method could potentially take a long time. Correct me if I am wrong, but ports 1-49151 are used by other applications. If testing if a port is available takes 100 ms, then testing 1-49151 would take over 1 hour. Perhaps it good to require the value in to be above 49151.

Thoughts?